src/Plugin/migrate/process/ApplyFilters.php

```diff
@@ -0,0 +1,316 @@
+    }
+  }
+
+  /**
+   * Validates the "filters_to_apply" and "filters_to_skip" configuration keys.
+   *
+   * @param string[] $keys
+   *   The configuration keys to validate.
+   */
+  protected function validateConfigurationKeyFiltersAndFiltersToSkip(array $keys): void {
+    $filter_format = $this->getConfigurationFilterFormat();
+    foreach ($keys as $key) {
+      if (array_key_exists($key, $this->configuration)) {
+        if (!is_array($this->configuration[$key]) || empty($this->configuration[$key])) {
+          throw new \InvalidArgumentException(sprintf('The configuration option "%s" must be a non-empty array.', $key));
+        }
+        else {
+          foreach ($this->configuration[$key] as $filter_id) {
+            if (!is_string($filter_id)) {
+              throw new \InvalidArgumentException(sprintf('The configuration option "%s" must contain strings. "%s" is not a string.', $key, var_export($filter_id, TRUE)));
+            }
+            if ($this->filterFormatManager->isValidFilterId($filter_id, $filter_format) !== TRUE) {
+              throw new \InvalidArgumentException(sprintf('"%s" is not a valid filter for the "%s" filter format.', $filter_id, $filter_format->id()));
+            }
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Returns the current filter format.
+   *
+   * @return \Drupal\filter\Entity\FilterFormat|null
+   *   The current filter format.
+   */
+  protected function getConfigurationFilterFormat(): ?FilterFormat {
+    return is_string($this->configuration['filter_format']) ? $this->filterFormatManager->getFilterFormat($this->configuration['filter_format']) : NULL;
+  }
+
+  /**
+   * Returns the destination language id.
+   *
+   * @return string
+   *   The destination language id or the site's default language id.
+   */
+  protected function getLangcode(): string {
+    return $this->row->getDestinationProperty('langcode') ?? $this->languageManager->getDefaultLanguage()->getId();
+  }
+
+  /**
+   * Applies a list of filters to a string.
+   *
+   * @param string $value
+   *   The string to filter.
+   *
+   * @return string
+   *   The filtered string.
+   */
+  protected function doFilter(string $value): string {
+    /** @var \Drupal\filter\FilterPluginCollection|null $filters */
+    $filters = $this->getFiltersToApply();
+    /** @var string $langcode */
+    $langcode = $this->getLangcode();
+    if ($filters && $filters->count() > 0) {
+      foreach ($filters as $filter) {
+        /** @var \Drupal\filter\FilterProcessResult $result */
+        $result = $filter->process($value, $langcode);
+        $value = $result->getProcessedText();
+      }
+    }
+
+    return $value;
+  }
+
+  /**
+   * Returns a list of filters to apply.
+   *
+   * @return \Drupal\filter\FilterPluginCollection|null
+   *   The list of filters to apply or null.
+   */
+  protected function getFiltersToApply(): ?FilterPluginCollection {
```

Comment:
As I see, this will always return a FilterPluginCollection no?

Reply:
**Status:** Fixed
Same as your previous comment on `getEnabledFilters()`.